dsgrid/registry/dimension_mapping_registry_manager.py

```diff
@@ -103,6 +107,27 @@ def check_unique_records(self, config: DimensionMappingsConfig, warn_only=False)
                     f"There are {len(duplicates)} dimension mappings with duplicate content (data files)."
                 )
 
+    def _check_records_against_dimension_records(self, config):
+        for mapping in config.model.mappings:
+            actual_from_records = {x.from_id for x in mapping.records}
+            actual_to_records = {x.to_id for x in mapping.records}
+            from_dimension = self._dimension_mgr.get_by_id(mapping.from_dimension.dimension_id)
+            allowed_from_records = from_dimension.get_unique_ids()
+            diff = actual_from_records.symmetric_difference(allowed_from_records)
+            if diff:
+                dim_id = from_dimension.model.dimension_id
+                raise DSGInvalidDimensionMapping(
+                    f"Dimension mapping has invalid 'from' records: dimension_id={dim_id} {diff}"
```

Comment:
"Dimension mapping has invalid and/or missing 'from' records: dimension_id={dim_id} {diff}"

Am I reading this correctly that diff will list both the invalid elements and the missing elements?

Reply:
`diff` will list the elements that are present in only one of the sets. The message needs to be corrected.